iree/turbine/kernel/wave/constraints.py

```diff
@@ -236,6 +238,52 @@ def apply(
                     1,  # N
                     1,  # K
                 ]
+            case MMAType.F32_16x16x32_K4_F8:
+                offset = [
+                    Piecewise(
+                        (lane % 16, ~MMA_ACC), (4 * floor(lane / 16), MMA_ACC)
+                    ),  # M
+                    lane % 16,  # N
+                    (16 * floor(GPR_NUM / 4))
```

Comment:
I can make it share a statement. 

WRT `index_sequence.partition(num_partitions = 2, ...)`. We actually have quite a few discussions on best way to represent this. I know Triton uses the num_partitions style, but from IREE-compiler side, after some discussions, we were using enum cases for now since it's easiest plumbing, and less prone to failure way.

Reply:
So right now you have to explicitly specify the new index sequence. What I am saying above is that maybe we can avoid that by defining an "algebra" over the index sequences that allows you to do things like expand the access pattern, partition the access pattern etc. with each one of these operators having well defined semantics. Yes, let's chat about it more.

Reply:
Makes sense, we can definitely do add some "algebra" to modify the existing index sequence based on the index sequence, shouldn't be too hard especially since we know what each constants mean! 

I think the hard part would probably be the design to layer it nicely and have a good way to control, decide when this modification will run.